src/toolbars/AudioSetupToolBar.cpp

```diff
@@ -0,0 +1,840 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSetupToolBar.cpp
+
+*******************************************************************//*!
+
+\class AudioSetupToolBar
+\brief A toolbar to allow easier changing of input and output devices .
+
+*//*******************************************************************/
+
+
+
+#include "AudioSetupToolBar.h"
```

Comment:
When you make a new .cpp/.h file pair, the .cpp should include the .h before any other inclusion, to be sure the .h file is really self-contained.

And you did that!  Was it accident or were you aware of this rule?